ql/pricingengines/swap/discretizedswap.cpp

```diff
@@ -19,8 +19,17 @@
 */
 
 #include <ql/pricingengines/swap/discretizedswap.hpp>
+#include <ql/settings.hpp>
 
 namespace QuantLib {
+    namespace {
+        inline bool useCouponInPostAdjust(const Time& resetTime,
+                                          const Time& payTime,
+                                          const boost::optional<bool>& includeTodaysCashFlows) {
+            return (resetTime < 0.0) &&
+                   ((payTime > 0.0) || (includeTodaysCashFlows && (payTime == 0.0)));
```

Comment:
I didn't remember `includeTodaysCashFlows` was an `optional<bool>`.  That means we have to write something like `includeTodaysCashFlows && * includeTodaysCashFlows` (the first checks that the `optional` contains a value, the second that the contained value is true).